Add plugin: Punch Clock

shunsuke1980 opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/shunsuke1980/obsidian-plugin-punch-clock

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[# Obsidian Punch Clock Plugin](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/README.md#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

["description": "Comprehensive time tracking plugin with real-time tracking, reporting, and CSV data storage.",](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/manifest.json#L6)
Don't include the word "plugin" in the description.

[const TIMER_ICON = `<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-timer"><line x1="10" x2="10" y1="2" y2="4"/><line x1="14" x2="14" y1="2" y2="4"/><line x1="12" x2="12" y1="14" y2="16"/><line x1="12" x2="10.5" y1="14" y2="5.5"/><circle cx="12" cy="14" r="8"/></svg>`;](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/main.ts#L27)
Obsidian already includes the lucide library, this is not necessary.

[showInRibbon: true,](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/main.ts#L18), [// Remove existing ribbon icon if it exists](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/main.ts#L109-L116), [// Show in ribbon](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/settings-tab.ts#L115-L127)
As of v1.1.0 of Obsidian, the ribbon can be customized within Obsidian. So users can choose what items to include and in what order all in one place. Because of this, we're recommending that new plugins avoid introducing their own toggles for ribbon items.


[name: 'Open Timer',](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/main.ts#L67), [name: 'Open Panel',](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/main.ts#L76), [name: 'Start Quick Timer',](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/main.ts#L85), [name: 'Stop Current Timer',](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/main.ts#L94), ['Punch Clock'](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/main.ts#L117), [contentEl.createEl('h2', { text: 'Edit Time Entry' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L39), [startTimeContainer.createEl('label', { text: 'Start Time' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L46), [endTimeContainer.createEl('label', { text: 'End Time' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L75), [contentEl.createEl('h2', { text: 'Start New Timer' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L233), [.setButtonText('Start Timer')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L272), [.setTooltip('Start New Timer')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L461), [headerEl.createEl('h3', { text: 'Timer Running' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L606), [.setButtonText('Stop Timer')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L675), [breakdownEl.createEl('h4', { text: 'Category Breakdown' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L737), [.setTooltip('Edit Entry')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L785), [.setTooltip('Delete Entry')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L793), [.setTooltip('Start New Timer with Same Settings')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L805), [breakdownEl.createEl('h4', { text: 'Category Breakdown' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L860), [dailyBreakdownEl.createEl('h3', { text: 'Daily Breakdown' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L902), [containerEl.createEl('h3', { text: 'Category Colors' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/settings-tab.ts#L57), [.setName('Default Category')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/settings-tab.ts#L87), [.setName('Auto-save Entries')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/settings-tab.ts#L106), [.setName('Default View')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/settings-tab.ts#L131), [.setName('Date Format')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/settings-tab.ts#L146), [.setName('Time Format')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/settings-tab.ts#L158), [.setName('Storage Directory')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/settings-tab.ts#L174), [.setName('Start Day of Week')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/settings-tab.ts#L186), [contentEl.createEl('h2', { text: 'Punch Clock' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/modal.ts#L28), [entriesContainer.createEl('h3', { text: 'Recent Entries' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/modal.ts#L253), [.setTooltip('Start New Timer')](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/modal.ts#L303)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI), product / brand names follow the casing set by the respective owner.

[import moment, { Moment } from 'moment';](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/view.ts#L4), 
Import the moment instance from the Obsidian API instead.

[containerEl.createEl('h2', { text: 'Punch Clock Settings' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/settings-tab.ts#L30)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[containerEl.createEl('h3', { text: 'Category Colors' });](https://github.com/shunsuke1980/obsidian-plugin-punch-clock/blob/4493e3dbb13f9541bb2f92fdb265be632c480b7e/src/settings-tab.ts#L57)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

shunsuke1980 commented:
Hi @joethei ,

Thank you for the feedback! I have addressed all the issues you mentioned:

✅ Removed "Obsidian" from the plugin header naming
✅ Updated description to remove the word "plugin" 
✅ Removed unnecessary Lucide library reference
✅ Updated ribbon settings as per v1.1.0 recommendations
✅ Fixed all UI text to use sentence case
✅ Updated moment.js import
✅ Fixed settings tab structure

The plugin repository has been updated and I've resolved the merge conflicts in this PR. Please let me know if there's anything else that needs to be adjusted.
